powerpc_syscall.H

```diff
@@ -0,0 +1,40 @@
+/******************************************************
+ * ArchC Architecture Dependent Syscall header file.  *
+ * This file is automatically generated by ArchC      *
+ * WITHOUT WARRANTY OF ANY KIND, either express       *
+ * or implied.                                        *
+ * For more information on ArchC, please visit:       *
+ * http://www.archc.org                               *
+ *                                                    *
+ * The ArchC Team                                     *
+ * Computer Systems Laboratory (LSC)                  *
+ * IC-UNICAMP                                         *
+ * http://www.lsc.ic.unicamp.br                       *
+ ******************************************************/
+ 
+
+#ifndef POWERPC_SYSCALL_H
+#define POWERPC_SYSCALL_H
+
+#include "powerpc_arch.H"
+#include "powerpc_arch_ref.H"
+#include "powerpc_parms.H"
+#include "ac_syscall.H"
+
+//powerpc system calls
+class powerpc_syscall : public ac_syscall<powerpc_parms::ac_word, powerpc_parms::ac_Hword>, public powerpc_arch_ref
+{
+public:
+  powerpc_syscall(powerpc_arch& ref) : ac_syscall<powerpc_parms::ac_word, powerpc_parms::ac_Hword>(ref, powerpc_parms::AC_RAMSIZE), powerpc_arch_ref(ref) {};
+  virtual ~powerpc_syscall() {};
+
+  void get_buffer(int argn, unsigned char* buf, unsigned int size);
+  void set_buffer(int argn, unsigned char* buf, unsigned int size);
+  void set_buffer_noinvert(int argn, unsigned char* buf, unsigned int size);
+  int  get_int(int argn);
+  void set_int(int argn, int val);
+  void return_from_syscall();
+  void set_prog_args(int argc, char **argv);
+};
+
+#endif
```

Comment:
ArchC follows the LLVM style code formatting. I recommend using the "clang-format" to help you.